osu.Game/Screens/OnlinePlay/Matchmaking/RankedPlay/DiscardScreen.cs

```diff
@@ -117,257 +106,87 @@ private void load()
                     ]
                 }
             ];
-
-            foreach (var (card, x) in layoutCards(cardFlow, -20))
-            {
-                card.X = x;
-                card.Y = cardYOffsetAt(x);
-                card.Rotation = cardRotationAt(x);
-            }
         }
 
-        private static float cardRotationAt(float x) => x * 0.03f;
-        private static float cardYOffsetAt(float x) => MathF.Pow(MathF.Abs(x / 250), 2) * 20;
-
-        private void cardClicked(Card target)
-        {
-            target.Selected = !target.Selected;
+        public override ICardFacadeContainer PlayerCardContainer => playerHand;
 
-            discardButton.Text = cardFlow.Any(it => it.Selected) ? "Discard" : "Don't Discard";
-        }
+        private double lastCardInsertionTime = double.MinValue;
+        private double lastDiscardTime = double.MinValue;
 
-        private void doDiscard()
+        private void onDiscardButtonClicked()
         {
             discardButton.Hide();
+            playerHand.SelectionMode = CardSelectionMode.Disabled;
 
-            foreach (var card in cards)
-                card.Action = null;
-
-            double delay = 0;
-
-            playDiscardAnimation(ref delay);
-            flyInNewCards(ref delay /*TODO: pass in newCards parameter */);
-            presentRemainingCards(ref delay);
-        }
-
-        #region Animations
-
-        private void playDiscardAnimation(ref double delay)
-        {
-            cards.RemoveAll(card => card.Selected);
-
-            var selectedCards = cardFlow.Where(it => it.Selected).ToList();
-
-            if (selectedCards.Count == 0)
-                return;
-
-            foreach (var (card, x) in layoutCards(selectedCards, 20))
-            {
-                card.Delay(delay)
-                    .MoveTo(new Vector2(x, -200), 400, Easing.OutExpo)
-                    .RotateTo(0, 400, Easing.OutExpo)
-                    .Then(400)
-                    .ScaleTo(0, 350)
-                    .FadeOut(200)
-                    .Expire();
-
-                delay += 50;
-            }
-
-            foreach (var (card, x) in layoutCards(cards, -20))
-            {
-                card
-                    .Delay(delay)
-                    .MoveToX(x, 400, Easing.OutExpo)
-                    .MoveToY(cardYOffsetAt(x), 400, Easing.OutExpo)
-                    .RotateTo(cardRotationAt(x), 400, Easing.OutExpo);
-
-                delay += 50;
-            }
-
-            delay += 1500;
+            Client.DiscardCards(playerHand.Selection.Select(it => it.Card).ToArray()).FireAndForget();
         }
 
-        private void flyInNewCards(ref double delay)
+        public override void CardAdded(RankedPlayScreen.Card card)
```

Comment:
This needs to somehow handle the incoming user. Was thinking either passing the user ID in here, or splitting it into `PlayerCardAdded` / `OpponentCardAdded`?